service-workers/service-worker/resources/test-helpers.sub.js

```diff
@@ -58,6 +58,25 @@ function with_iframe(url) {
     });
 }
 
+function with_sandboxed_iframe(url, sandbox) {
```

Comment:
If you add an "options" dictionary to with_iframe to support auto_remove, you could also add a sandbox option to that same dictionary rather than having a separate function for this.